From ef81bcbda40f7cf8f6d38d199bdd876e69bd9d7f Mon Sep 17 00:00:00 2001
From: Matt Ellis <matell@microsoft.com>
Date: Mon, 20 Mar 2017 14:07:50 -0700
Subject: [PATCH 2/3] Add core specific build scripts.

`build-core.cmd` and `build-core.sh` don't depend on any internal
resources.
---
 build-core.cmd      |  5 +++++
 build-core.sh       | 13 +++++++++++++
 build.sh            | 13 -------------
 tools/Build.sh      | 13 -------------
 tools/BuildCore.cmd | 19 +++++++++++++++++++
 tools/BuildCore.sh  | 13 +++++++++++++
 6 files changed, 50 insertions(+), 26 deletions(-)
 create mode 100644 build-core.cmd
 create mode 100755 build-core.sh
 delete mode 100755 build.sh
 delete mode 100755 tools/Build.sh
 create mode 100644 tools/BuildCore.cmd
 create mode 100755 tools/BuildCore.sh

diff --git a/build-core.cmd b/build-core.cmd
new file mode 100755
index 0000000..4e01691
--- /dev/null
+++ b/build-core.cmd
@@ -0,0 +1,5 @@
+@ECHO OFF
+
+PowerShell -NoProfile -NoLogo -ExecutionPolicy unrestricted -Command "[System.Threading.Thread]::CurrentThread.CurrentCulture = ''; [System.Threading.Thread]::CurrentThread.CurrentUICulture = '';& '%~dp0dotnet-install.ps1' %*; exit $LASTEXITCODE"
+SET PATH=%localappdata%\Microsoft\dotnet;%PATH%
+tools\BuildCore.cmd
diff --git a/build-core.sh b/build-core.sh
new file mode 100755
index 0000000..4aaec8c
--- /dev/null
+++ b/build-core.sh
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+set -euo pipefail
+IFS=$'\n\t'
+
+SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")"
+DOTNET_TOOL_ROOT="$SCRIPT_ROOT/obj/tools/dotnet"
+
+curl -L https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh | bash -s -- --version 1.0.0 --install-dir "$DOTNET_TOOL_ROOT"
+
+export PATH=$DOTNET_TOOL_ROOT:$PATH
+
+$SCRIPT_ROOT/tools/BuildCore.sh
diff --git a/build.sh b/build.sh
deleted file mode 100755
index 2ccb664..0000000
--- a/build.sh
+++ /dev/null
@@ -1,13 +0,0 @@
-#!/bin/bash
-
-set -euo pipefail
-IFS=$'\n\t'
-
-SCRIPT_ROOT="$(dirname "${BASH_SOURCE[0]}")"
-DOTNET_TOOL_ROOT="$SCRIPT_ROOT/obj/tools/dotnet"
-
-curl -L https://raw.githubusercontent.com/dotnet/cli/rel/1.0.0/scripts/obtain/dotnet-install.sh | bash -s -- --version 1.0.0 --install-dir "$DOTNET_TOOL_ROOT"
-
-export PATH=$DOTNET_TOOL_ROOT:$PATH
-
-$SCRIPT_ROOT/tools/Build.sh
diff --git a/tools/Build.sh b/tools/Build.sh
deleted file mode 100755
index 30ae273..0000000
--- a/tools/Build.sh
+++ /dev/null
@@ -1,13 +0,0 @@
-#!/bin/bash
-set -euo pipefail
-IFS=$'\n\t'
-
-SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-
-. "$SCRIPT_ROOT/EnsureWebSdkEnv.sh"
-
-dotnet restore "$WebSdkRoot/src/Publish/Microsoft.NET.Sdk.Publish.Tasks/Microsoft.NET.Sdk.Publish.Tasks.csproj" /p:Configuration=Release
-
-dotnet build "$WebSdkRoot/src/Publish/Microsoft.NET.Sdk.Publish.Tasks/Microsoft.NET.Sdk.Publish.Tasks.csproj" /p:Configuration=Release
-
-dotnet pack "$WebSdkRoot/build.dotnet.proj" /p:Configuration=Release
diff --git a/tools/BuildCore.cmd b/tools/BuildCore.cmd
new file mode 100644
index 0000000..8a1b7e6
--- /dev/null
+++ b/tools/BuildCore.cmd
@@ -0,0 +1,19 @@
+@echo off
+
+call %~dp0\EnsureWebSdkEnv.cmd
+
+call dotnet restore %WebSdkRoot%src\Publish\Microsoft.NET.Sdk.Publish.Tasks\Microsoft.NET.Sdk.Publish.Tasks.csproj /p:Configuration=Release
+if errorlevel 1 GOTO ERROR
+
+call dotnet build %WebSdkRoot%src\Publish\Microsoft.NET.Sdk.Publish.Tasks\Microsoft.NET.Sdk.Publish.Tasks.csproj /p:Configuration=Release
+if errorlevel 1 GOTO ERROR
+
+call dotnet pack %WebSdkRoot%build.dotnet.proj /p:Configuration=Release
+if errorlevel 1 GOTO ERROR
+
+endlocal
+
+exit /b 0
+
+:ERROR
+exit /b 1
diff --git a/tools/BuildCore.sh b/tools/BuildCore.sh
new file mode 100755
index 0000000..30ae273
--- /dev/null
+++ b/tools/BuildCore.sh
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+
+SCRIPT_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+. "$SCRIPT_ROOT/EnsureWebSdkEnv.sh"
+
+dotnet restore "$WebSdkRoot/src/Publish/Microsoft.NET.Sdk.Publish.Tasks/Microsoft.NET.Sdk.Publish.Tasks.csproj" /p:Configuration=Release
+
+dotnet build "$WebSdkRoot/src/Publish/Microsoft.NET.Sdk.Publish.Tasks/Microsoft.NET.Sdk.Publish.Tasks.csproj" /p:Configuration=Release
+
+dotnet pack "$WebSdkRoot/build.dotnet.proj" /p:Configuration=Release
-- 
1.8.3.1

